meilisearch-release-changelog/v1.12.0.md

```diff
@@ -0,0 +1,137 @@
+
+<!-- The following lines should NOT be put in the PRE-release changelogs -->
+🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.
+
+<!-- The following lines should NOT be put in the PRE-release changelogs -->
+Some SDKs might not include all new features. Consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).
+
+# New features and updates 🔥
+
+## Improve indexing speed
+
+Indexing time for huge dataset import (multiple millions of documents) is divided by two!
+
+TBD: give more data and metrics about improvements
+
+Done by @dureuill, @ManyTheFish, and @Kerollmops in #4900.
+
+## New index settings: use `facetSearch` and `prefixSearch` to improve indexing speed
+
+v1.12 introduces two new index settings: `facetSearch` and `prefixSearch`.
+
+Both settings allow you to skip parts of the indexing process. This leads to significant improvements to indexing speed, but may negatively impact search experience in some use cases.
+
+Done by @ManyTheFish in #5091
+
+### `facetSearch`
+
+Use this setting to toggle [facet search](https://www.meilisearch.com/docs/learn/filtering_and_sorting/search_with_facet_filters#searching-facet-values):
+
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/books/settings/facet-search' \
+  -H 'Content-Type: application/json' \
+  --data-binary 'true'
+```
+
+The default value for `facetSearch` is `true`. When set to `false`, this setting disables facet search for all filterable attributes in an index.
+
+### `prefixSearch`
+
+Use this setting to configure the ability to [search a word by prefix](https://www.meilisearch.com/docs/learn/engine/prefix) on an index:
+
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/books/settings/prefix-search' \
+  -H 'Content-Type: application/json' \
+  --data-binary 'disabled'
+```
+
+`prefixSearch` accepts one of the following values:
+
+- `"indexingTime"`: enables prefix processing during indexing. This is the default Meilisearch behavior
+- `"disabled"`: deactivates prefix search completely
+
+
+Disabling prefix search means the query `he` will no longer match the word `hello`. This may significantly impact search result relevancy, but speeds up the indexing process.
+
+## New API route: `/batches`
+
+The new `/batches` endpoint allow you to query information about task batches.
+
+`GET` `/batches` returns a list of batch objects:
+
+```sh
+curl  -X GET 'http://localhost:7700/batches'
+```
+
+This endpoint accepts the same parameters as `GET` `/tasks` route, allowing you to narrow down which batches you want to see.
+
```

Comment:
```suggestion

This endpoint accepts the same parameters and returns the same error codes as the `GET` `/tasks` route, allowing you to narrow down which batches you want to see.
But don't forget, the filters are applied **to the tasks**, not the batches. Which means, querying `GET /batches?status=failed` won't return the batches that failed but will instead returns the batches **containing at least** one failed task.
The same is true for all query parameters. (`uids` specify the `uids` of the tasks, `before_started_at`, etc)
And your API key also impacts the results. A batch won't be returned if it only contains tasks your API key don't let you see.
```

Reply:
@irevoire is the suggestion of @guimachiavelli correct to you?

Reply:
Hey,

> For some parameters like status=failed I guess it's irrelevant, since AFAIK the whole batch fails if even a single task fails.

No! That's not the case, a single batch can contain both failed and succeeded tasks.
And also, a batch doesn't have a status by itself at all.
This means that you can match the same batch by using the filter `?statuses=succeeded` and `?statuses=failed`.

> I would definitely expect GET /batches?uids=0,2 to return batches with a uid of either 1 or 2, and even more so if GET /tasks?uids=0,2 returns tasks 1 and 2.

Yeah, I too think that's a bit confusing.
The purpose of this naming was that you can copy paste your filter on the `/tasks` route and it just works. There is no « oh I must change the `uid` in `batchUid` », or the `uid` to `taskUid`.
That's why I intentionally chose this example in my message hehe

And yes I think your suggestion is better than my initial message, thanks :+1:

Reply:
Ok, using gui's version then!